sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/maskExpressions.scala

```diff
@@ -23,9 +23,54 @@ import org.apache.spark.sql.catalyst.analysis.TypeCheckResult.DataTypeMismatch
 import org.apache.spark.sql.catalyst.expressions.codegen._
 import org.apache.spark.sql.catalyst.expressions.codegen.Block._
 import org.apache.spark.sql.errors.QueryErrorsBase
-import org.apache.spark.sql.types.{AbstractDataType, DataType, StringType}
+import org.apache.spark.sql.types.{AbstractDataType, DataType, IntegerType, StringType}
 import org.apache.spark.unsafe.types.UTF8String
 
+/**
+ * The trait does Input Data Type validation .
```

Comment:
please also mention which types of expressions we intend to inherit this trait, perhaps with a couple examples?

Reply:
Moved more common code into this trait from Masking classes  and mentioned additional detail in the description